Static Analysis and Security Testing adjustments (#499)
cncf/tag-security: security-whitepaper/cloud-native-security-whitepaper.md

##### Static Analysis and Security Testing

> Static analysis of IaC, application manifests, and software code provides linting and identification of misconfigurations and vulnerabilities. IaC code should be subject to the same pipeline policy controls as are application workloads.

Reviewer: It’s important to note that static analysis of software code will examine only that specific component (microservice for example) without any context of the surrounding components. Therefore, the ability of such a tool to provide accurate identification of vulnerabilities across the application is very limited. Unfortunately, such a static analysis will result in a high rate FN and FP in a cloud-native environment (if we want to be more specific - in a microservices architecture)

Reviewer 2: @RonVider  i think the distinction is important enough to call it out.  made update
```suggestion
Static analysis of IaC, application manifests, and software code provide linting, identification of misconfigurations, and vulnerabilities of the specific component and will not include context of the surrounding components in analysis.  It is important to note that individual analysis is important, but should not be the only form of analysis. IaC code should be subject to the same pipeline policy controls as are application workloads.
```

> Static analysis of IaC, application manifests, and software code provide linting, identification of misconfigurations, and vulnerabilities of the specific component and will not include context of the surrounding components in analysis.  It is important to note that individual analysis is important, but should not be the only form of analysis. IaC code should be subject to the same pipeline policy controls as are application workloads.

Reviewer 3: > It is important to note that individual analysis is important, but should not be the only form of analysis.

I think that this is a bit too much of a detail for the whitepaper. This is more a consideration for the implementation of the tools rather than the operator implementing these tooling. 

> IaC code should be subject to the same pipeline policy controls as are application workloads.

I would prefer the term similar since it is not going to be exactly the same as workloads.

Reviewer 2: ```suggestion
Static analysis of IaC, application manifests, and software code provide linting, identification of misconfigurations, and vulnerabilities of the specific component and will not include context of the surrounding components in analysis.  It is important to note that individual analysis is important, but should not be the only form of analysis. IaC code should be subject to the similar pipeline policy controls as are application workloads.
```

> IaC is an increasingly popular way for organizations to deploy cloud and container infrastructure. Insecure configurations in IaC templates will naturally lead to vulnerabilities in the deployed infrastructure. These templates should therefore be scanned for characteristics that compromise security, using automated tools, before deploying the application and infrastructure artifacts. Key misconfigurations to keep an eye out for include:

Reviewer 3: ```suggestion
IaC is an increasingly popular way for organizations to deploy cloud and container infrastructure. Insecure configurations in IaC templates will naturally lead to vulnerabilities in the deployed infrastructure. These templates should therefore be scanned for characteristics that compromise security, before deploying the application and infrastructure artifacts. Key misconfigurations to keep an eye out for include:
```

Already mentioned earlier about making it a pipeline. I think the idea of having a secure process here is more of the focus than the automation of it.
